### DISCUSSION TITLE
Android build app failed due to add firebaseappdistribution code in gradle.build release file

### DISCUSSION THREAD

**Discussion by gsazu** [codemagic-ci-cd/codemagic-docs#1717] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Android (Kotlin)

### Steps to reproduce

I followed all steps for firebase app distribution in codemagic.yaml file.
But, when I add inside the buildTypes and release
        firebaseAppDistribution {
            serviceCredentialsFile=System.getenv('GOOGLE_APPLICATION_CREDENTIALS')
      }

the build not performed. is says at terminal Could not find method firebaseAppDistribution()
also when I run it through codemagic, build not performed

### Expected results

the video on your website told us the old solution like firebase_token, but not supported now by codemagic. 
Need to generate build.

### Actual results

Build apk or aab file will be created and published

**Comment by dtrdic** [comment]
Hey @local-create, let's check this first,
1. have you added the app in firebase console? On the project settings > general > Your apps

**Reply by dtrdic** [reply, marked as answer]
2. In your **root-level (project-level)** Gradle file (<project>/build.gradle or <project>/build.gradle.kts), add the App Distribution Gradle plugin as a buildscript dependency:

```
buildscript {
  repositories {
    // Make sure that you have the following two repositories
    google()  // Google's Maven repository
    mavenCentral()  // Maven Central repository
  }

  dependencies {
      ...
      classpath("com.android.tools.build:gradle:7.2.0")

      // Make sure that you have the Google services Gradle plugin dependency
      classpath("com.google.gms:google-services:4.3.15")

      // Add the dependency for the App Distribution Gradle plugin
      classpath("com.google.firebase:firebase-appdistribution-gradle:4.0.0")
  }
}
```


3. In your module (app-level) Gradle file (usually on android/app/build.gradle, add the App Distribution Gradle plugin:

```
apply plugin: 'com.google.firebase.appdistribution'
```
